geometry/optimization/graph_of_convex_sets.h

```diff
@@ -31,6 +31,11 @@ struct GraphOfConvexSetsOptions {
   relaxation is tight. */
   bool convex_relaxation{true};
 
+  /** Performs a preprocessing step to remove edges that cannot lie on the
+  path from source to target. In most cases, preprocessing causes a net
+  reduction in computation by reducing the size of the optimization solved. */
+  bool preprocessing{true};
```

Comment:
This has been discussed above but, if can say my opinion, I would make the preprocessing inactive by default, since as Mark said in many problems it substantially slow down the solve. I think that for the maze problem this slow down is quite dramatic (~10x), at least it was with the python overhead. I'd imagine a user being more annoyed by realizing that we are running a preprocessing that slows things down, rather than realizing that he/she can decrease the runtimes by turning on an extra feature that we provide?